crates/ark/src/lsp/statement_range.rs

```diff
@@ -92,105 +90,246 @@ fn new_statement_range_response(
+            // right before we found `@examples` or `@examplesIf`. That's the start of
+            // our node range.
+            if sibling_text.starts_with("@examples") || sibling_text.starts_with("@examplesIf") {
+                start = Some(last_sibling);
+            }
 
-        // Have we exited roxygen comments specifically?
-        if !RE_ROXYGEN2_COMMENT.is_match(sibling) {
-            return None;
+            break;
         }
 
-        let sibling = RE_ROXYGEN2_COMMENT.replace(sibling, "").into_owned();
+        last_sibling = sibling;
+    }
+
+    let Some(start) = start else {
+        // No `@examples` or `@examplesIf` found
+        return None;
+    };
 
-        // Trim off any leading whitespace
-        let sibling = sibling.trim_start();
+    last_sibling = node;
 
-        // Did we discover that the `node` was indeed in `@examples`?
-        if sibling.starts_with("@examples") {
-            code = Some(text.to_string());
+    // Walk "down" the page
+    //
+    // Goal is to find the last line in this `@examples` or `@examplesIf` section
+    while let Some(sibling) = last_sibling.next_sibling() {
+        // Have we exited roxygen comments?
+        let Some(sibling_text) = as_roxygen_comment_text(&sibling, contents) else {
             break;
-        }
+        };
+
+        // Drop `#'` from the comment's text
+        let sibling_text = RE_ROXYGEN2_COMMENT.replace(&sibling_text, "");
 
-        // Otherwise, did we discover the `node` was in a different tag?
-        if sibling.starts_with("@") {
+        // Trim leading whitespace after the `#'`
+        let sibling_text = sibling_text.trim_start();
+
+        // Did we discover a new tag?
+        if sibling_text.starts_with("@") {
             break;
         }
+
+        last_sibling = sibling;
     }
 
-    return Some((node, code));
+    let end = last_sibling;
+
+    let range = tree_sitter::Range {
+        start_byte: start.start_byte(),
+        end_byte: end.end_byte(),
+        start_point: start.start_position(),
+        end_point: end.end_position(),
+    };
+
+    return Some(range);
+}
+
+fn find_roxygen_examples_range(
+    root: &Node,
+    range: tree_sitter::Range,
+    contents: &Rope,
+    point: Point,
+) -> Option<(tree_sitter::Range, String)> {
+    // Anchor row that we adjust relative to
+    let row_adjustment = range.start_point.row;
+
+    // Slice out the `@examples` or `@examplesIf` code block (with leading roxygen comments)
+    let Some(slice) = contents.get_byte_slice(range.start_byte..range.end_byte) else {
+        return None;
+    };
+
+    // Trim out leading roxygen comments so we are left with a subdocument of actual code
+    let subcontents = slice.to_string();
+    let subcontents: Vec<String> = subcontents
+        .lines()
+        .map(|line| {
+            // Trim `#'` and at most 1 leading whitespace character. Don't trim more
```

Comment:
The space trimming is also incorrect in strings (again, a reasonable limitation)